rewrite-maven/src/main/java/org/openrewrite/maven/search/ModuleHasDependency.java

```diff
@@ -0,0 +1,120 @@
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class ModuleHasDependency extends ScanningRecipe<ModuleHasDependency.Accumulator> {
+
+    @Override
+    public String getDisplayName() {
+        return "Module has dependency";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Searches for Maven modules that have a dependency matching the specified groupId and artifactId. " +
+               "Places a `SearchResult` marker on all sources within a module with a matching dependency. " +
+               "This recipe is intended to be used as a precondition for other recipes. " +
+               "For example this could be used to limit the application of a spring boot migration to only projects " +
+               "that use spring-boot-starter, limiting unnecessary upgrading. " +
+               "If the search result you want is instead just the build.gradle(.kts) file applying the plugin, use the `FindDependency` recipe instead.";
+    }
+
+    @Option(displayName = "Group",
+            description = "The first part of a dependency coordinate `com.google.guava:guava:VERSION`. Supports glob.",
+            example = "com.google.guava")
+    String groupId;
+
+    @Option(displayName = "Artifact",
+            description = "The second part of a dependency coordinate `com.google.guava:guava:VERSION`. Supports glob.",
+            example = "guava")
+    String artifactId;
+
+    @Option(displayName = "Version",
+            description = "An exact version number or node-style semver selector used to select the version number.",
+            example = "3.0.0",
+            required = false)
+    @Nullable
+    String version;
+
+    @Option(displayName = "Version pattern",
+            description = "Allows version selection to be extended beyond the original Node Semver semantics. So for example," +
+                          "Setting 'version' to \"25-29\" can be paired with a metadata pattern of \"-jre\" to select Guava 29.0-jre",
+            example = "-jre",
+            required = false)
+    @Nullable
+    String versionPattern;
+
+    @Value
+    public static class Accumulator {
+        Set<JavaProject> projectsWithDependency;
+    }
+
+    @Override
+    public Accumulator getInitialValue(ExecutionContext ctx) {
+        return new Accumulator(new HashSet<>());
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getScanner(Accumulator acc) {
+        return new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
+                assert tree != null;
+                tree.getMarkers()
+                        .findFirst(JavaProject.class)
+                        .ifPresent(jp -> {
+                            Tree t = new FindDependency(groupId, artifactId, version, versionPattern).getVisitor().visit(tree, ctx);
```

Comment:
Did you possibly want to use `maven.DependencyInsight` here like was done with the Gradle variation?

Reply:
Good call, will update.

Reply:
Updated!